### PR TITLE
analyze the codebase with implicit-casts: false

### PR DESCRIPTION
**Pull request by devoncarew**
- analyze the codebase with `implicit-casts: false`

This mode will be part of NNBD, so is an incremental step in that direction (converting the codebase over supporting NNBD).

### CODE REVIEW

**Comment by devoncarew** [comment]
I don't see the CI issues locally, but perhaps deleting my analysis driver cache will reset things?

**Comment by devoncarew** [comment]
Ah, I believe this is because the analysis_options file sits above the project directories, so the analysis server doesn't see file changes. Re-starting shows the implicit cast issues.

**Comment by devoncarew** [comment]
Hmm, it looks like there are ~400 issues with implicit casts in the codebase. I changed this PR to disable the setting, and addressed perhaps 20 of the issues (as well as adding types to public method calls).

File: packages/devtools/example/test.dart [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Test {
     final List<int> numbers = List.generate(10, (index) => index * index);
     // ignore: unused_local_variable
     final Map<int, String> numberDescriptions =
-        Map.fromIterable(numbers, value: (key) => _toWord(key));
+        Map.fromIterable(numbers, value: (key) => _toWord(key as int));
```
Comment (natebosch) [inline packages/devtools/example/test.dart]: [optional] If you're on a recent SDK:

```dart
{for (var n in numbers) _toWord(n): n};
```

Comment (devoncarew) [reply packages/devtools/example/test.dart]: 👍 done

**Pull request merged**